app/controllers/recipes_controller.rb

```diff
@@ -1,31 +1,52 @@
-class RecipesController < ApplicationController
-  def index
-    @recipes = Recipe.all
-  end
-
-  # def show
-  #   @recipe = Recipe.find(:params[id])
-  # end
-
-  def new; end
-
-  def create
-    @recipe = Recipe.new(recipe_params)
-    if @recipe.save
-      redirect_to recipe_path, notice: 'Recipe created successfully!'
-    else
-      render new, status: :unprocessable_entity
-    end
-  end
-
-  def public_recipes
-    @public_recipes = Recipe.includes(:user).all.where(public: true).order(created_at: :desc)
-    render :public_recipes
-  end
-
-  private
-
-  def recipe_params
-    params.permit(:name, :preparation_time, :cooking_time, :description, :public)
-  end
-end
+class RecipesController < ApplicationController
+  def index
+    @user = current_user
+    @recipes = @user.recipes
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    data = Recipe.where(id: params[:id])
+    @recipe = data[0]
+    @inventory_foods = Food.all.joins('INNER JOIN recipe_foods ON foods.id = recipe_foods.food_id')
+      .order(created_at: :desc).select('foods.*, recipe_foods.quantity, recipe_foods.id as recipe_foods_id')
+      .where(recipe_foods: { recipe_id: params[:id] })
+    return unless current_user
+
+    @inventory_hash = Inventory.all.where(user_id: current_user.id).map { |inventory| [inventory.name, inventory.id] }

```

Comment:
Again I didnt understand why this is here? Recipes controller should render recipes not a map of inventory table. Also show action should render single recipe.